### ISSUE TITLE
[Analyzer] Domaincheck

### ISSUE THREAD

**Issue by mlodic** [intelowlproject/IntelOwl#1682]
## Name
Domaincheck

## Link

https://github.com/zuzazuza/domaincheck

## Type of analyzer
domain only

## Why should we use it
cool tool to check DNS related problems

## Possible implementation
integrate the open source tool: https://github.com/zuzazuza/domaincheck

**Comment by g4ze** [comment]
hi! @mlodic  since the tool seems like a standalone command, it cant be imported as a package after cloning and installing. Could you suggest some workaround for it?

**Comment by mlodic** [comment]
yes the tool is pretty raw but it is still much useful and clever.

One idea, that I have already done before in such cases, is that we could create a package for this and maintain it.

I created a fork [here](https://github.com/intelowlproject/domaincheck). What you could do is to add the necessary items to publish it on pypi and then use the pypi package in this analyzer.

It's something fast to do once you know what to touch. Check what I did [here](https://github.com/mlodic/pdfid) for pdfid. I added the github workflow, managed the version and setup.py configuration and that's all.

Thoughts?

**Comment by g4ze** [comment]
That might be a bit of an overkill imo, cant we just `import subprocess` and use the shell command to directly fetch results, we could install it sipmly via the dockerfile..? Is that going to increase the processing time significantly?

**Comment by g4ze** [comment]
my bad, accidental click

**Comment by mlodic** [comment]
I honestly do not like too much and would prefer to avoid it if possibile.  Anyway, even if we went in that direction, we would still have a problem with the installation. We cannot make the binary directly in our image (too expensive) and, even if we build it beforehand, we cannot embed it in IntelOwl itself without a proper installation method (that would mean distribution -> license implications).

For some python code, even if they are a little, makes sense to do what I suggested and it is really easy and plain, it's not an overkill. Have you ever created a python package before? That is also a chance to try to do that.

**Comment by g4ze** [comment]
okay, i understand, it makes more sense now. Installation would expensive in our case. Let's do it the way you proposed!
Yes I've published a small package `key-proxy` .

**Comment by g4ze** [comment]
referencing latest [update](https://github.com/intelowlproject/domaincheck/pull/1)

**Comment by mlodic** [comment]
I have just finalized the creation of the package: https://pypi.org/project/domaincheck-fork/ you can now use it to complete the issue :)

**Comment by g4ze** [comment]
Woohoo!

**Comment by g4ze** [comment]
hey @mlodic  after importing `domaincheck`, I'm not able to access its modules or anything. Are we expected to run it as a command?

**Comment by mlodic** [comment]
the package is `domaincheck-fork`, are you sure to have used the correct one?

**Comment by g4ze** [comment]
after installing the  published package `pip install domaincheck-fork`:

![Image](https://github.com/user-attachments/assets/fcd9ccca-4866-4311-9e77-e64fe4e3ea70)
Neither of the import are working. 
I'm not able to access its modules. the command `domaincheck` is functional. I might be missing something here, but i have deleted all the domaincheck packages and did a fresh reinstall of `domaincheck-fork`

**Comment by mlodic** [comment]
ok that's happening cause the python script name does not end with .py............
I fixed it, pls try again with version 0.2

**Comment by g4ze** [comment]
there's still a slight problem with approach we are taking, domaincheck is a standalone script and we cant use its modiles as it is. we need to make the functions inside more modular and reusable outside. I'll try to fix that and make a PR in the fork. Please let me know if what I'm saying is valid.

**Comment by g4ze** [comment]
referencing the proposed [fix](https://github.com/intelowlproject/domaincheck/pull/2)
this works as expected, not the best solution but gets the work done.

**Comment by mlodic** [comment]
we noticed that the tool was not feasible to be integrated
